Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -415,10 +415,10 @@ class SourceGenerator {
         let createIntermediateGroups = targetSource.createIntermediateGroups ?? project.options.createIntermediateGroups
 
         let directories = children
-            .filter { $0.isDirectory && $0.extension == nil && $0.extension != "lproj" }
+            .filter { $0.isDirectory && $0.extension != "lproj" && $0.extension != "xcassets" && $0.extension != "bundle" }
 
         let filePaths = children
-            .filter { $0.isFile || $0.extension != nil && $0.extension != "lproj" }
+            .filter { $0.isFile || $0.extension != nil && $0.extension != "lproj" && ($0.extension == "xcdatamodeld" || !$0.isDirectory) }
```

Comment:
This is to make sure that the directory doesn't get added as a file

If we don't have this if check, then the generated project has two references of the folder, one in Compile Sources and one in Copy Bundle Resources phase

If we don't have the check for `xcdatamodeld` a unit test related to versionGroup checks starts failing, probably because it won't be able to find `.xccurrentversion` if the directory is not considered in the file path.